src/python_pachyderm/pps_client.py

```diff
@@ -45,11 +43,11 @@ def inspect_job(self, job_id, block_state=False):
     def list_job(self, pipeline=None, input_commit=None, output_commit=None):
         if isinstance(input_commit, list):
             input_commit = [commit_from(ic) for ic in input_commit]
-        else:
+        elif isinstance(input_commit, str):
             input_commit = [commit_from(input_commit)]
         if output_commit:
             output_commit = commit_from(output_commit)
-        req = proto.ListJobRequest(pipeline=pipeline, input_commit=input_commit, output_commit=output_commit)
+        req = proto.ListJobRequest(pipeline=proto.Pipeline(name=pipeline), input_commit=input_commit, output_commit=output_commit)
```

Comment:
Since name is the only field for pipelines, this makes sense. But could you rename the field to `pipeline_name`? That'll be more consistent with naming conventions elsewhere, and reduce the chances of type errors from codebases using the old function signature.